nixos/modules/services/hardware/fwupd.nix

```diff
@@ -187,13 +187,18 @@ in {
       # fwupd-refresh expects a user that we do not create, so just run with DynamicUser
       # instead and ensure we take ownership of /var/lib/fwupd
       services.fwupd-refresh.serviceConfig = {
-        DynamicUser = true;
```

Comment:
This could probably be made to work with some more options but I personally find this super hard to debug and get to work so I can totally understand if people just ditch it for a normal user.